### PR TITLE
Add descriptions to auxiliary modules Actions

### LINKED ISSUE #13403: Suggestion: remove "Actions" from module with only one action (which is also the default one) to prevent confusion
## Steps to reproduce

1. `use auxiliary/server/dhcp`
2. `show actions`

## Expected behavior

There shouldn't be any "action" for module with only one action (which is also the default one)

## Current behavior

We see the "Service" action but... What does it do? Should I set it? Why is there only one? I think it's confusing...

```terminal
msf5 auxiliary(server/dhcp) > show actions

Auxiliary actions:

   Name     Description
   ----     -----------
   Service  
```

Here is the action creation:
https://github.com/rapid7/metasploit-framework/blob/4a853beb8d8be4369cdef67d3d655385d1edfa00/modules/auxiliary/server/dhcp.rb#L20-L28

But it doesn't seem to be used afterwards.

Several other modules are concerned (non-exhaustive list), for example "ftp" or "fakedns" under auxiliary/server

Let me know what you think, and if you like it I can propose a PR to clean it.

### PR DESCRIPTION
**Pull request by cnotin**
And a little formatting
Closes #13403

I encourage you to check the auxiliary module example file as I changed it too, and it's important that it reflects how you prefer them to look like ;)

## Verification

- [x] Start `msfconsole`
- [x] `use psnuffle`
- [x] `show actions`
- [x] **Verify** that you see action descriptions
- [x] **Verify** that the syntax isn't broken anywhere (I didn't try to load each modified module but I guess the automatic CI/CD job here would catch it isn't it?)

### CODE REVIEW

**Comment by cnotin** [comment]
Travis caught a syntax error :)

**Comment by wvu-r7** [comment]
This PR will help a lot of people who are confused about poorly named actions!

File: modules/auxiliary/gather/coldfusion_pwd_props.rb [hunk 1]
```diff
@@ -29,12 +29,6 @@ def initialize(info = {})
           'nebulus'
         ],
       'License'        => MSF_LICENSE,
-      'Actions'     =>
```
Comment (wvu-r7) [inline modules/auxiliary/gather/coldfusion_pwd_props.rb:32]: Please update the module description!

File: modules/auxiliary/gather/firefox_pdfjs_file_theft.rb [hunk 2]
```diff
@@ -24,7 +24,7 @@ def initialize(info={})
         'Unknown'  # Metasploit module
       ],
       'License'     => MSF_LICENSE,
-      'Actions'     => [[ 'WebServer' ]],
+      'Actions'        => [[ 'WebServer', 'Description' => 'Serve exploit via webserver' ]],
```
Comment (wvu-r7) [inline modules/auxiliary/gather/firefox_pdfjs_file_theft.rb:27]:
```suggestion
      'Actions'        => [[ 'WebServer', 'Description' => 'Serve exploit via web server' ]],
```

File: modules/auxiliary/gather/flash_rosetta_jsonp_url_disclosure.rb [hunk 3]
```diff
@@ -34,7 +34,7 @@ def initialize(info = {})
           ['URL', 'http://quaxio.com/jsonp_handcrafted_flash_files/']
         ],
       'DisclosureDate' => 'Jul 8 2014',
-      'Actions'        => [ [ 'WebServer' ] ],
+      'Actions'        => [[ 'WebServer', 'Description' => 'Serve exploit via webserver' ]],
```
Comment (wvu-r7) [inline modules/auxiliary/gather/flash_rosetta_jsonp_url_disclosure.rb:37]:
```suggestion
      'Actions'        => [[ 'WebServer', 'Description' => 'Serve exploit via web server' ]],
```

File: modules/auxiliary/gather/samsung_browser_sop_bypass.rb [hunk 4]
```diff
@@ -28,7 +28,7 @@ def initialize(info = {})
         ['URL', 'http://fr.0day.today/exploit/description/28434']
         ],
         'DisclosureDate' => 'Nov 08 2017',
-        'Actions'        => [[ 'WebServer' ]],
+        'Actions'        => [[ 'WebServer', 'Description' => 'Serve exploit via webserver' ]],
```
Comment (wvu-r7) [inline modules/auxiliary/gather/samsung_browser_sop_bypass.rb:31]:
```suggestion
        'Actions'        => [[ 'WebServer', 'Description' => 'Serve exploit via web server' ]],
```

File: modules/auxiliary/server/capture/http.rb [hunk 5]
```diff
@@ -19,7 +19,7 @@ def initialize
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'Capture' ]
+          [ 'Capture', 'Description' => 'Run capture webserver' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/http.rb:22]:
```suggestion
          [ 'Capture', 'Description' => 'Run capture web server' ]
```

File: modules/auxiliary/server/capture/http_basic.rb [hunk 6]
```diff
@@ -24,7 +24,7 @@ def initialize(info={})
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'Capture' ]
+          [ 'Capture', 'Description' => 'Run capture webserver' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/http_basic.rb:27]:
```suggestion
          [ 'Capture', 'Description' => 'Run capture web server' ]
```

File: modules/auxiliary/server/capture/http_ntlm.rb [hunk 7]
```diff
@@ -28,7 +28,7 @@ def initialize(info = {})
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'WebServer' ]
+          [ 'WebServer', 'Description' => 'Run capture webserver' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/http_ntlm.rb:31]:
```suggestion
          [ 'WebServer', 'Description' => 'Run capture web server' ]
```

File: modules/auxiliary/server/capture/printjob_capture.rb [hunk 8]
```diff
@@ -29,7 +29,7 @@ def initialize
         # Readers for resulting PCL/PC
         ['URL', 'http://www.ghostscript.com']
       ],
-      'Actions'        => [[ 'Capture' ]],
+      'Actions'        => [[ 'Capture', 'Description' => 'Run Printjob capture server' ]],
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/printjob_capture.rb:32]:
```suggestion
      'Actions'        => [[ 'Capture', 'Description' => 'Run print job capture server' ]],
```

File: modules/auxiliary/server/dns/spoofhelper.rb [hunk 9]
```diff
@@ -22,7 +22,7 @@ def initialize
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'Service' ]
+          [ 'Service', 'Description' => 'Run DNS capture server' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/dns/spoofhelper.rb:25]:
```suggestion
          [ 'Service', 'Description' => 'Run DNS spoofing server' ]
```

File: modules/auxiliary/server/http_ntlmrelay.rb [hunk 10]
```diff
@@ -45,7 +45,7 @@ def initialize(info = {})
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'WebServer' ]
+          [ 'WebServer', 'Description' => 'Start webserver waiting for incoming authenticated connections' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/http_ntlmrelay.rb:48]:
```suggestion
          [ 'WebServer', 'Description' => 'Start web server waiting for incoming authenticated connections' ]
```

File: modules/auxiliary/server/local_hwbridge.rb [hunk 11]
```diff
@@ -25,7 +25,7 @@ def initialize(info = {})
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'WebServer' ]
+          [ 'WebServer', 'Description' => 'Run bridge webserver' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/local_hwbridge.rb:28]:
```suggestion
          [ 'WebServer', 'Description' => 'Run HWBridge web server' ]
```

File: modules/auxiliary/server/socks4a.rb [hunk 12]
```diff
@@ -17,7 +17,7 @@ def initialize
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'Proxy' ]
+          [ 'Proxy', 'Description' => 'Run socks4a proxy' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/socks4a.rb:20]:
```suggestion
          [ 'Proxy', 'Description' => 'Run SOCKS4a proxy' ]
```

File: modules/auxiliary/server/socks5.rb [hunk 13]
```diff
@@ -20,7 +20,7 @@ def initialize
       'License'        => MSF_LICENSE,
       'Actions'        =>
         [
-          [ 'Proxy' ]
+          [ 'Proxy', 'Description' => 'Run socks5 proxy' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/socks5.rb:23]:
```suggestion
          [ 'Proxy', 'Description' => 'Run SOCKS5 proxy' ]
```

File: modules/auxiliary/server/socks_unc.rb [hunk 14]
```diff
@@ -20,7 +20,7 @@ def initialize
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'Proxy' ]
+          [ 'Proxy', 'Description' => 'Run SOCKS proxy' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/socks_unc.rb:23]:
```suggestion
          [ 'Proxy', 'Description' => 'Run SOCKS UNC proxy' ]
```

File: modules/auxiliary/server/webkit_xslt_dropper.rb [hunk 15]
```diff
@@ -23,7 +23,7 @@ def initialize(info = {})
         ],
       'Actions'     =>
         [
-          [ 'WebServer' ]
+          [ 'WebServer', 'Description' => 'Serve exploit via webserver' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/webkit_xslt_dropper.rb:26]:
```suggestion
          [ 'WebServer', 'Description' => 'Serve exploit via web server' ]
```

File: modules/auxiliary/client/iec104/iec104.rb [hunk 16]
```diff
@@ -22,9 +22,9 @@ def initialize(info = {})
         ],
       'License'        => MSF_LICENSE,
       'Actions'        =>
-      [
-        ['SEND_COMMAND', { 'Description' => 'Send command to device' }]
-      ],
+       [
+         ['SEND_COMMAND', { 'Description' => 'Send command to device' }]
+       ],
```
Comment (wvu-r7) [inline modules/auxiliary/client/iec104/iec104.rb:27]:
```suggestion
        [
          ['SEND_COMMAND', 'Description' => 'Send command to device']
        ],
```

File: modules/auxiliary/gather/browser_info.rb [hunk 17]
```diff
@@ -25,8 +25,8 @@ def initialize(info={})
               'Description' => 'A web that collects information about the browser.'
           }]
         ],
-      'PassiveActions' => [ 'WebServer' ],
-      'DefaultAction'  => 'WebServer'
+        'PassiveActions' => [ 'WebServer' ],
+        'DefaultAction'  => 'WebServer'
```
Comment (wvu-r7) [inline modules/auxiliary/gather/browser_info.rb:29]:
```suggestion
      'PassiveActions' => [ 'WebServer' ],
      'DefaultAction'  => 'WebServer'
```

File: modules/auxiliary/gather/browser_info.rb [hunk 18]
```diff
@@ -25,8 +25,8 @@ def initialize(info={})
               'Description' => 'A web that collects information about the browser.'
```
Comment (wvu-r7) [inline modules/auxiliary/gather/browser_info.rb:25]:
```suggestion
              'Description' => 'A web server that collects information about the browser.'
```

File: modules/auxiliary/server/capture/drda.rb [hunk 19]
```diff
@@ -37,7 +37,7 @@ def initialize
       },
       'Author'         => 'Patrik Karlsson <patrik[at]cqure.net>',
       'License'        => MSF_LICENSE,
-      'Actions'        => [ [ 'Capture' ] ],
+      'Actions'        => [[ 'Capture', 'Description' => 'Run capture server' ]],
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/drda.rb:40]:
```suggestion
      'Actions'        => [[ 'Capture', 'Description' => 'Run DRDA capture server' ]],
```

File: modules/auxiliary/server/capture/ftp.rb [hunk 20]
```diff
@@ -18,7 +18,7 @@ def initialize
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'Capture' ]
+          [ 'Capture', 'Description' => 'Run capture server' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/ftp.rb:21]:
```suggestion
          [ 'Capture', 'Description' => 'Run FTP capture server' ]
```

File: modules/auxiliary/server/capture/mssql.rb [hunk 21]
```diff
@@ -37,7 +37,7 @@ def initialize
       },
       'Author'         => 'Patrik Karlsson <patrik[at]cqure.net>',
       'License'        => MSF_LICENSE,
-      'Actions'        => [ [ 'Capture' ] ],
+      'Actions'        => [ [ 'Capture', 'Description' => 'Run MSSQL capture server' ] ],
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/mssql.rb:40]:
```suggestion
      'Actions'        => [[ 'Capture', 'Description' => 'Run MSSQL capture server' ]],
```

File: modules/auxiliary/server/capture/mysql.rb [hunk 22]
```diff
@@ -17,7 +17,7 @@ def initialize
       },
       'Author'         => 'Patrik Karlsson <patrik[at]cqure.net>',
       'License'        => MSF_LICENSE,
-      'Actions'        => [ [ 'Capture' ] ],
+      'Actions'        => [ [ 'Capture', 'Description' => 'Run MySQL capture server' ] ],
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/mysql.rb:20]:
```suggestion
      'Actions'        => [[ 'Capture', 'Description' => 'Run MySQL capture server' ]],
```

File: modules/auxiliary/server/capture/pop3.rb [hunk 23]
```diff
@@ -19,7 +19,7 @@ def initialize
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'Capture' ]
+          [ 'Capture' , 'Description' => 'Run POP3 capture server']
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/pop3.rb:22]:
```suggestion
          [ 'Capture' , 'Description' => 'Run POP3 capture server' ]
```

File: modules/auxiliary/server/capture/postgresql.rb [hunk 24]
```diff
@@ -15,7 +15,7 @@ def initialize
         capture clear-text authentication credentials.},
       'Author'         => 'Dhiru Kholia <dhiru[at]openwall.com>',
       'License'        => MSF_LICENSE,
-      'Actions'        => [ [ 'Capture' ] ],
+      'Actions'        => [ [ 'Capture', 'Description' => 'Run PostgreSQL capture server' ] ],
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/postgresql.rb:18]:
```suggestion
      'Actions'        => [[ 'Capture', 'Description' => 'Run PostgreSQL capture server' ]],
```

File: modules/auxiliary/server/capture/sip.rb [hunk 25]
```diff
@@ -18,7 +18,7 @@ def initialize
       },
       'Author'         => 'Patrik Karlsson <patrik[at]cqure.net>',
       'License'        => MSF_LICENSE,
-      'Actions'        => [ [ 'Capture' ] ],
+      'Actions'        => [ [ 'Capture', 'Description' => 'Run SIP capture server' ] ],
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/sip.rb:21]:
```suggestion
      'Actions'        => [[ 'Capture', 'Description' => 'Run SIP capture server' ]],
```

File: modules/auxiliary/server/capture/smb.rb [hunk 26]
```diff
@@ -28,9 +28,9 @@ def initialize
       },
       'Author' => 'hdm',
       'License' => MSF_LICENSE,
-      'Actions' => [ [ 'Sniffer' ] ],
-      'PassiveActions' => [ 'Sniffer' ],
-      'DefaultAction' => 'Sniffer'
+      'Actions' => [ [ 'Capture', 'Description' => 'Run SMB capture server' ] ],
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/smb.rb:31]:
```suggestion
      'Actions' => [[ 'Capture', 'Description' => 'Run SMB capture server' ]],
```

File: modules/auxiliary/server/capture/telnet.rb [hunk 27]
```diff
@@ -20,7 +20,7 @@ def initialize
       },
       'Author'         => 'kris katterjohn',
       'License'        => MSF_LICENSE,
-      'Actions'        => [ [ 'Capture' ] ],
+      'Actions'        => [ [ 'Capture', 'Description' => 'Run telnet capture server' ] ],
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/telnet.rb:23]:
```suggestion
      'Actions'        => [[ 'Capture', 'Description' => 'Run telnet capture server' ]],
```

File: modules/auxiliary/server/capture/vnc.rb [hunk 28]
```diff
@@ -16,7 +16,7 @@ def initialize
       },
       'Author'         => 'Patrik Karlsson <patrik[at]cqure.net>',
       'License'        => MSF_LICENSE,
-      'Actions'        => [ [ 'Capture' ] ],
+      'Actions'        => [ [ 'Capture', 'Description' => 'Run VNC capture server' ] ],
```
Comment (wvu-r7) [inline modules/auxiliary/server/capture/vnc.rb:19]:
```suggestion
      'Actions'        => [[ 'Capture', 'Description' => 'Run VNC capture server' ]],
```

File: modules/auxiliary/server/fakedns.rb [hunk 29]
```diff
@@ -20,7 +20,7 @@ def initialize
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'Service' ]
+          [ 'Service', 'Description' => 'Run DNS service' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/fakedns.rb:23]:
```suggestion
          [ 'Service', 'Description' => 'Run DNS server' ]
```

File: modules/auxiliary/server/tftp.rb [hunk 30]
```diff
@@ -20,7 +20,7 @@ def initialize
       'License'     => MSF_LICENSE,
       'Actions'     =>
         [
-          [ 'Service' ]
+          [ 'Service', 'Description' => 'Run TFTP server to serve files' ]
```
Comment (wvu-r7) [inline modules/auxiliary/server/tftp.rb:23]:
```suggestion
          [ 'Service', 'Description' => 'Serve files via TFTP' ]
```

**Review by wvu — CHANGES_REQUESTED** [review]
Minor changes for consistency.

**Comment by wvu-r7** [comment]
Well, that was a slog. I'm sure you can relate!

File: modules/auxiliary/gather/coldfusion_pwd_props.rb (hunk 1 above)
Comment (cnotin) [reply modules/auxiliary/gather/coldfusion_pwd_props.rb:32]: Oh yes indeed good catch. I'll do!

File: modules/auxiliary/server/capture/printjob_capture.rb (hunk 8 above)
Comment (wvu-r7) [reply modules/auxiliary/server/capture/printjob_capture.rb:32]: There's one more here.

**Comment by wvu-r7** [comment]
I'll squash the commits when I land. No worries.

File: modules/auxiliary/gather/browser_info.rb (hunk 18 above)
Comment (wvu-r7) [reply modules/auxiliary/gather/browser_info.rb]: I don't think this was applied.

File: modules/auxiliary/gather/coldfusion_pwd_props.rb (hunk 1 above)
Comment (cnotin) [reply modules/auxiliary/gather/coldfusion_pwd_props.rb:32]: Done in 0085b2c93e4f6a3a84355ad641bfd130ac61b91b

File: modules/auxiliary/gather/browser_info.rb (hunk 18 above)
Comment (cnotin) [reply modules/auxiliary/gather/browser_info.rb]: No because I applied the previous on same file and now the diff is "outdated". I'll commit it manually

Comment (cnotin) [reply modules/auxiliary/gather/browser_info.rb]: Done in dde207731a8abc19eba8e76e5bfa6a1aeba60569

**Comment by cnotin** [comment]
All good and linear I think :)

File: modules/auxiliary/gather/browser_info.rb [hunk 31]
```diff
@@ -22,7 +22,7 @@ def initialize(info={})
         [
           [
             'WebServer', {
```
Comment (wvu-r7) [inline modules/auxiliary/gather/browser_info.rb:24]:
```suggestion
            'WebServer',
```

File: modules/auxiliary/gather/browser_info.rb [hunk 32]
```diff
@@ -22,7 +22,7 @@ def initialize(info={})
...
-              'Description' => 'A web that collects information about the browser.'
+              'Description' => 'A web server that collects information about the browser.'
           }]
```
Comment (wvu-r7) [inline modules/auxiliary/gather/browser_info.rb:26]:
```suggestion
          ]
```

**Review by wvu — APPROVED** [review]
Waiting for build to pass, then landing. Modules load fine.

**Comment by wvu-r7** [comment]
# Release Notes

This adds or updates action descriptions for numerous auxiliary and post modules in order to improve the user experience when listing or choosing actions.

**Comment by cnotin** [comment]
Cool!

**Comment by wvu-r7** [comment]
Sanity test hasn't passed in two hours.

**Comment by cnotin** [comment]
That can happens or it's certainly bad?

**Comment by wvu-r7** [comment]
The sanity tests run in serial for every push across every PR... They haven't been parallelized yet.

**Pull request merged**